py/strato/common/log/log2text.py

```diff
@@ -122,51 +178,29 @@ def _getNextParsableEntry(inputStream, logFile, colorCode, formatter):
     list the file until the next parsable line
     finish when all lines were listed
     """
+    logTypeConf = formatter._getLogTypeConf(logFile)
     while True:
         try:
             line = inputStream.next()
-            obj = json.loads(line)
-            formatted = formatter.process(obj)
-            return obj, None if formatted is None else _addLogName(formatted, colorCode, logFile)
+            formatted, timestamp = formatter.process(line, logFile, logTypeConf)
+            return None if formatted is None else _addLogName(formatted, colorCode, logFile), timestamp
         except StopIteration:
             return None
         except:
-            line = line.strip()
-            timestampMatch = re.match(r'(\d+\.\d+)(.*)', line)
-            if timestampMatch is not None:
-                timestamp = float(timestampMatch.group(1).split('.')[0])
-                formatted = "%s %s" % (formatter._absoluteClock(timestamp), timestampMatch.group(2))
-                return {'created': timestamp}, _addLogName(formatted, colorCode, logFile)
-            else:
-                dateMatch = re.match(r'(\d{4})/(\d{2})/(\d{2}) (\d{2}):(\d{2}):(\d{2})', line)
-                if dateMatch is not None:
-                    dt = datetime.datetime(int(dateMatch.group(1)), int(dateMatch.group(2)), int(dateMatch.group(3)),
-                                           int(dateMatch.group(4)), int(dateMatch.group(5)), int(dateMatch.group(6)))
-                    timestamp = time.mktime(dt.timetuple())
-                    return {'created': timestamp}, _addLogName(line, colorCode, logFile)
-                else:
-                    return {'created': 0}, line
-            # print "Failed to parse record '%s' " % line
-
+            return line, HIGHEST_PRIORITY
 
 def _getColorCode(id):
     return MULTY_LOG_COLORS[id % (len(MULTY_LOG_COLORS) - 1)]
 
 
 def printLogs(logFiles, formatter):
-    inputStreams = [(open(logFile), logFile) for logFile in logFiles]
+    inputStreams = [(open(logFile, 'r'), logFile) for logFile in logFiles]
 
-    # initialize current lines
-    currentLines= []
-    for streamId, (inputStream, logFile) in enumerate(inputStreams):
-        currentLines.append(_getNextParsableEntry(inputStream, logFile, _getColorCode(streamId), formatter))
-
-    while True:
-        # finished all input streams
-        if not any(currentLines):
-            break
+    currentLines= [_getNextParsableEntry(inputStream, logFile, _getColorCode(streamId), formatter)
+                   for streamId, (inputStream, logFile) in enumerate(inputStreams)]
 
-        _, nextStreamId, formatted = min((line[0]['created'], streamId, line[1])
+    while any(currentLines):
+        _, nextStreamId, formatted = min((line[1], streamId, line[0])
```

Comment:
i've removed the ['created'] part, as it was passing back the object (and i return the epoch timestamp instead)
i can't remove more than that, as it is required to chose which stream's last entry has the lowest timestamp to know what to print next